lib/node.js

```diff
@@ -211,22 +210,36 @@ class Node {
       case 'role':
         delete this.kuzzle.repositories.role.roles[data.id];
         break;
-      case 'strategy:added':
-        try {
-          this.kuzzle.pluginsManager.registerStrategy(data.pluginName, data.name, data.strategy);
-        } catch (e) {
-          // log & discard
-          this.context.log('error', `Plugin ${data.pluginName} - tried to add the strategy "${data.name}": ${e.message}`);
-        }
-        break;
-      case 'strategy:removed':
-        try {
-          this.kuzzle.pluginsManager.unregisterStrategy(data.pluginName, data.name);
-        } catch (e) {
-          // log & discard
-          this.context.log('error', `Plugin ${data.pluginName} - tried to remove the strategy "${data.name}": ${e.message}`);
-        }
-        break;
+      case 'strategies':
+        return this.redis.hgetall('cluster:strategies')
+          .then(response => {
+            const currentStrategies = new Set(this.kuzzle.pluginsManager.listStrategies());
+
+            const strategies = response || {};
+
+            for (const name of Object.keys(strategies)) {
```

Comment:
Why do not use `for ... in ...`? Not a judgment, I do not know the differences?

Reply:
With `for..in` you have to check if the returned property belongs to this object instance, and not inherited.

So the pattern when using `for...in` is:

```js
for (const name in strategies) {
  if (strategies.hasOwnProperty(name)) {
    /* ... */
  }
}
```

So, not really better than using `for...of` with `Object.keys`.
Performance-wise, I didn't check on recent Node.js versions, but `for...in` was known to be very slow.

Reply:
Thank you :slightly_smiling_face: